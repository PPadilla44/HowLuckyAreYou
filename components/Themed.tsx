import { Text as DefaultText, View as DefaultView, ScrollViewProps as dScrollViewProps } from 'react-native';
import { Icon as DefaultIcon, IconProps as DefaultIconProps, Input as DefaultInput, InputProps as DefaultInputProps, ButtonGroup as DButtonGroup, ButtonGroupProps as DButtonGroupProps } from 'react-native-elements';
import { ScrollView as DefaultScrollView, NativeViewGestureHandlerProps, TouchableOpacity as DefaultTouchableOpacity } from "react-native-gesture-handler";

import Colors from '../constants/Colors';

import useColorScheme from '../hooks/useColorScheme';

export const useThemeColor = (
    props: { light?: string; dark?: string },
    colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) => {
    // const theme = useColorScheme();
    const theme = "dark";

    const colorFromProps = props[theme];

    return { colorFromProps: colorFromProps ? colorFromProps : Colors[theme][colorName], theme }
}

type ThemeProps = {
    lightColor?: string;
    darkColor?: string;
};

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];
export type TouchableOpacityProps = ThemeProps & DefaultTouchableOpacity['props'];
export type IconProps = ThemeProps & DefaultIconProps;
export type InputProps = ThemeProps & DefaultInputProps;
export type ScrollViewProps = ThemeProps & NativeViewGestureHandlerProps & dScrollViewProps;
export type ButtonGroupProps = ThemeProps & DButtonGroupProps;

export const Text = (props: TextProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;

    return <DefaultText style={[{ color, fontFamily: "Futura" }, style]} {...otherProps} />
}

export const View = (props: ViewProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    const shadowColor = useThemeColor({  }, "text").colorFromProps;
    return <DefaultView style={[{ backgroundColor, shadowColor }, style]}  {...otherProps} />
}

export const TouchableOpacity = (props: TouchableOpacityProps) => {
    const { style, containerStyle, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "button").colorFromProps;
    const oppositeColor = useThemeColor({ }, "text").colorFromProps;
    return <DefaultTouchableOpacity style={[{ backgroundColor, borderColor: oppositeColor, shadowColor: oppositeColor }, style]} containerStyle={[{ backgroundColor, borderColor: oppositeColor }, containerStyle]} {...otherProps} />
}

export const Icon = (props: IconProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "icon").colorFromProps;
    return <DefaultIcon tvParallaxProperties={undefined} color={color} {...otherProps} />
}

export const Input = (props: InputProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const { colorFromProps, theme } = useThemeColor({ light: lightColor, dark: darkColor }, "input");
    const backgroundColor = colorFromProps;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;
    return (

        <>
            <DefaultInput
                style={[{ backgroundColor: "transparent", color }, style]}
                keyboardAppearance={theme}
                {...otherProps}
            />
            <View style={{ backgroundColor, width: "100%", height: "100%", opacity: .5, position: "absolute", borderRadius: 15, zIndex: -10 }} />
        </>
    )
    // return <DefaultInput style={[{backgroundColor: "transparent", color }, style]} keyboardAppearance={theme} {...otherProps} />
}

export const ScrollView = (props: ScrollViewProps) => {
    const { style, darkColor, lightColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    return <DefaultScrollView style={[{ backgroundColor }, style]} {...otherProps} />
}

export const ButtonGroup = (props: ButtonGroupProps) => {
    const { containerStyle, darkColor, lightColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "input").colorFromProps;
    
    return <DButtonGroup containerStyle={[{ backgroundColor: color, borderColor: color }, containerStyle]}  {...otherProps} />
}