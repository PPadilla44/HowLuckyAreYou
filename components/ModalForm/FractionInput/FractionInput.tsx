import { StyleSheet } from 'react-native';
import React, { FC, useState } from 'react';
import { TextAsIcon } from '../../UI';
import { TouchableOpacity, View, Text, Input } from '../../Themed';
import Colors from '../../../constants/Colors';
import MultiplierModal from './MultiplierModal';


export interface Props {
    denominator: string;
    numerator: string;
    handleChanges: (data: {}) => void;
    multiplier: string;
}

const FractionInput: FC<Props> = ({ denominator, numerator, handleChanges, multiplier }) => {


    const [modalVisible, setModalVisible] = useState(false);


    return (
        <>

            <View style={{ flex: 2, backgroundColor: "transparent" }}>
                <Input
                    testID='numeratorInput'
                    keyboardType="number-pad"
                    style={styles.input}
                    inputContainerStyle={styles.inputContainer}
                    placeholder='1'
                    maxLength={2}
                    value={numerator}
                    onChangeText={numerator => handleChanges({ numerator })}
                />
            </View>
            <View style={{ marginHorizontal: 10, backgroundColor: "transparent", height: 52 }}>
                <TextAsIcon style={{ fontSize: 48, fontWeight: "bold" }} text={'/'} />
            </View>
            <View style={{ flex: 3, backgroundColor: "transparent" }}>
                <Input
                    testID='denominatorInput'
                    keyboardType="number-pad"
                    style={styles.input}
                    inputContainerStyle={styles.inputContainer}
                    placeholder='10'
                    maxLength={4}
                    value={denominator}
                    onChangeText={denominator => handleChanges({ denominator })}
                />
            </View>
            <TouchableOpacity darkColor={Colors.light.input} lightColor={Colors.dark.modal}
                onPress={() => setModalVisible(true)}
                containerStyle={{
                    alignSelf: "center",
                    alignItems: "center",
                    justifyContent: "center",
                    height: 35,
                    marginHorizontal: 10,
                    borderRadius: 5,
                    width: 52,
                }}
            >
                <Text darkColor={Colors.dark.modal} lightColor={Colors.light.input} style={{ fontSize: 18, fontWeight: "bold" }}>x {multiplier}</Text>
            </TouchableOpacity>

            <MultiplierModal
                modalVisible={modalVisible}
                setModalVisible={setModalVisible}
                mult={multiplier}
                handleChanges={handleChanges}
            />

        </>
    )
}

export default FractionInput

const styles = StyleSheet.create({
    input: {
        borderRadius: 15,
        fontSize: 48,
        fontWeight: "bold",
        fontFamily: "Futura",
        textAlign: "center",
        height: 52,
        overflow: "visible"
    },
    inputContainer: {
        borderBottomWidth: 0,
        overflow: "visible"
    },


})