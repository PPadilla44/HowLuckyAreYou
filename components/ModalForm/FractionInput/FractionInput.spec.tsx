import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import { TouchableOpacity, View } from "../../Themed";
import FractionInput, { Props } from "./FractionInput";

describe('<FractionInput />', () => {
    
    const mockFn = jest.fn();

    const props : Props = {
        denominator: "1",
        numerator: "5",
        handleChanges: mockFn,
        multiplier: "1"
    }

    let wrapper: RenderAPI;

    afterEach(cleanup)

    beforeEach(() => {
        wrapper = render(<FractionInput {...props} /> )
    });

    it('should render correctly', () => {
        expect(wrapper.container.children).toHaveLength(5);
        expect(wrapper.container.findAllByType(View)).toHaveLength(7);
        expect(wrapper.container.findAllByType(TouchableOpacity)).toHaveLength(1);
    });

    it('should render correct placeholder and values', () => {
        wrapper.getByPlaceholderText("1");
        wrapper.getByPlaceholderText("10");
        wrapper.getByDisplayValue("5");
        wrapper.getByDisplayValue("1");
    });

    it('should call handleChanges on textChange', () => {
        const numInput = wrapper.getByPlaceholderText("1");
        const denInput = wrapper.getByPlaceholderText("10");
        fireEvent.changeText(numInput, "testing1");
        fireEvent.changeText(denInput, "testing2");
        expect(mockFn).toBeCalledTimes(2)
    });

})