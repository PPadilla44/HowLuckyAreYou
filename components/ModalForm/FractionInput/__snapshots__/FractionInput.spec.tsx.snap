// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FractionInput /> should match snapshot 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "shadowColor": "white",
        },
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardAppearance="dark"
          keyboardType="number-pad"
          maxLength={2}
          onChangeText={[Function]}
          placeholder="1"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 15,
              "color": "white",
              "flex": 1,
              "fontFamily": "Futura",
              "fontSize": 48,
              "fontWeight": "bold",
              "height": 52,
              "minHeight": 40,
              "textAlign": "center",
            }
          }
          testID="numeratorInput"
          underlineColorAndroid="transparent"
          value="5"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "shadowColor": "white",
          },
          Object {
            "backgroundColor": "#808080",
            "borderRadius": 15,
            "height": "100%",
            "opacity": 0.5,
            "position": "absolute",
            "width": "100%",
            "zIndex": -10,
          },
        ]
      }
    />
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "shadowColor": "white",
        },
        Object {
          "backgroundColor": "transparent",
          "height": 52,
          "marginHorizontal": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Futura",
          },
          Object {
            "fontSize": 48,
            "fontWeight": "bold",
          },
        ]
      }
    >
      /
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "black",
          "shadowColor": "white",
        },
        Object {
          "backgroundColor": "transparent",
          "flex": 2,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          keyboardAppearance="dark"
          keyboardType="number-pad"
          maxLength={3}
          onChangeText={[Function]}
          placeholder="10"
          placeholderTextColor="#86939e"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 15,
              "color": "white",
              "flex": 1,
              "fontFamily": "Futura",
              "fontSize": 48,
              "fontWeight": "bold",
              "height": 52,
              "minHeight": 40,
              "textAlign": "center",
            }
          }
          testID="denominatorInput"
          underlineColorAndroid="transparent"
          value="1"
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "black",
            "shadowColor": "white",
          },
          Object {
            "backgroundColor": "#808080",
            "borderRadius": 15,
            "height": "100%",
            "opacity": 0.5,
            "position": "absolute",
            "width": "100%",
            "zIndex": -10,
          },
        ]
      }
    />
  </View>,
  <RNGestureHandlerButton
    collapsable={false}
    exclusive={true}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "backgroundColor": "#E6E6E6",
          "borderColor": "white",
        },
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderRadius": 5,
          "height": 35,
          "justifyContent": "center",
          "marginHorizontal": 10,
          "width": 52,
        },
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "backgroundColor": "#E6E6E6",
          "borderColor": "white",
          "opacity": 1,
          "shadowColor": "white",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#252525",
              "fontFamily": "Futura",
            },
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
      >
        BIL
      </Text>
    </View>
  </RNGestureHandlerButton>,
]
`;
